server/vcs/provider/github/converter/push_event.go

```diff
@@ -42,9 +42,20 @@ func (p PushEvent) Convert(e *github.PushEvent) (event.Push, error) {
 
 	installationToken := githubapp.GetInstallationIDFromEvent(e)
 
+	action := event.UpdatedAction
+
+	if e.GetCreated() {
+		action = event.CreatedAction
+	}
+
+	if e.GetDeleted() {
+		action = event.DeletedAction
+	}
+
 	return event.Push{
-		Repo: repo,
-		Sha:  e.GetHeadCommit().GetSHA(),
+		Repo:   repo,
+		Sha:    e.GetHeadCommit().GetID(),
```

Comment:
According to the [docs](https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#pushevent), if we want the 
`The SHA of the most recent commit on ref after the push` we can use `e.GetHead()` but I'm guessing we should only be using fields populated by the webhook event? Is that why we're using the `HeadCommit` field instead of `Ref`?

Reply:
I don't even see that in the github event payload tbh

Reply:
but I see headcommit